// react-router-app/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _core = require("@material-ui/core");

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// simple header
var Header = function Header() {
  var history = (0, _reactRouterDom.useHistory)();

  var onClickHandler = function onClickHandler() {
    console.log('clicked on button to go to about page');
    history.push('/aboutt');
  };

  return _react.default.createElement("nav", null, _react.default.createElement("ul", null, _react.default.createElement("li", null, _react.default.createElement(_reactRouterDom.Link, {
    to: "/homee"
  }, "Home")), _react.default.createElement("li", null, _react.default.createElement(_core.Button, {
    onClick: onClickHandler
  }, "to About")), _react.default.createElement("li", null, _react.default.createElement(_reactRouterDom.Link, {
    to: "/userss"
  }, "Users"))));
};

function ReactRouterComponent() {
  return _react.default.createElement(_reactRouterDom.BrowserRouter, null, _react.default.createElement("div", null, _react.default.createElement(Header, null), _react.default.createElement(_reactRouterDom.Switch, null, _react.default.createElement(_reactRouterDom.Route, {
    path: "/aboutt"
  }, _react.default.createElement(About, null)), _react.default.createElement(_reactRouterDom.Route, {
    path: "/userss"
  }, _react.default.createElement(Users, null)), _react.default.createElement(_reactRouterDom.Route, {
    path: "/homee"
  }, _react.default.createElement(Home, null)))));
}

function Home() {
  return _react.default.createElement("h2", null, "Home");
}

function About() {
  return _react.default.createElement("h2", null, "About");
}

function Users() {
  return _react.default.createElement("h2", null, "Users");
}

var _default = ReactRouterComponent;
exports.default = _default;
